### PR TITLE
UI: Added Spectator UI panel

### PR DESCRIPTION
**Pull request by Alf21**
image

Based on #112 and #111

### CODE REVIEW

**Comment by Westsaid** [comment]
I really like that you've put the name of the spectated player in the left bottom corner. We might want to change the color, i'll try to look up a suiting color, something like a yellow/orange color.

**Comment by Alf21** [comment]
> I really like that you've put the name of the spectated player in the left bottom corner. We might want to change the color, i'll try to look up a suiting color, something like a yellow/orange color.

Color is black for ROLE.None and the role color if you are aware of the player's role

File: code/ui/deadhud/SpectatedPlayerInfo.cs [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+            public override void Tick()
+            {
+                base.Tick();
+
+                if (_currentPlayer == null)
+                {
+                    return;
+                }
+
+                if (_currentPlayer.Role is NoneRole)
```
Comment (mzegar) [inline code/ui/deadhud/SpectatedPlayerInfo.cs:73]: Can`NoneRole` just contain the colour?

File: code/ui/deadhud/SpectatedPlayerInfo.cs [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+                    Style.BackgroundColor = _currentPlayer.Role.Color;
+                }
+
+                Style.Dirty();
+
+                _nameLabel.Text = $"{_currentPlayer.GetClientOwner()?.Name}";
+            }
+        }
+
+        private class IndicatorsPanel : Panel
```
Comment (mzegar) [inline code/ui/deadhud/SpectatedPlayerInfo.cs:88]: This is a copy and paste of how we do the health HUD on an alive player right? Is there any way to just reuse the code?

File: code/ui/deadhud/SpectatedPlayerInfo.cs (hunk 1 above)
Comment (Alf21) [reply code/ui/deadhud/SpectatedPlayerInfo.cs:73]: It already has a color, but not the color we want (transparent).

File: code/ui/deadhud/SpectatedPlayerInfo.cs (hunk 2 above)
Comment (Alf21) [reply code/ui/deadhud/SpectatedPlayerInfo.cs:88]: We could put it into a shared folder, but that's quite ugly. Additionally, we don't know what we gonna change in the future. One thing we could do is to use the same scss.

**Review by mzegar — APPROVED** [review]
LGTM

**Pull request merged**